### PR TITLE
LPS-194843 Adapting ObjectValidationRuleSettings to accept the new Object Unique Key Validation Rule Settings

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://liferay.atlassian.net/browse/LPS-194843

### CODE REVIEW

**Review by pedro-oliveira446 — APPROVED** [review]

**Comment by pedro-oliveira446** [comment]
Hi @mateussgn, great Job!

Just a note. I believe you need to run the baseline in the object-api module.

**Comment by mateussgn** [comment]
Sure!
Thanks for the reminder, @pedro-oliveira446!

**Comment by mateussgn** [comment]
ci:test:relevant

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectValidationRuleResourceImpl.java [hunk 1]
```diff
@@ -333,6 +333,27 @@ private ObjectValidationRule _toObjectValidationRule(
 					return serviceBuilderObjectValidationRuleSetting;
 				}
 
+				if (StringUtil.equals(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectValidationRuleResourceImpl.java]: Can we create a private method to reuse this logic? maybe something like replace or transform property name
So you would just need to pass two Strings, for example: ercPropertyName, idPropertyName
case 1: `NAME_OUTPUT_OBJECT_FIELD_EXTERNAL_REFERENCE_CODE, NAME_OUTPUT_OBJECT_FIELD_ID`
case 2: `NAME_KEY_OBJECT_FIELD_EXTERNAL_REFERENCE_CODE, NAME_KEY_OBJECT_FIELD_ID`

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/converter/ObjectValidationRuleDTOConverter.java [hunk 2]
```diff
@@ -129,10 +130,18 @@ private ObjectValidationRuleSetting _toObjectValidationRuleSetting(
...
-						if (!objectValidationRuleSetting.compareName(
+						if (FeatureFlagManagerUtil.isEnabled("LPS-187854") &&
+							!(objectValidationRuleSetting.compareName(
 								ObjectValidationRuleSettingConstants.
-									NAME_KEY_OBJECT_FIELD_ID) ||
-							!objectValidationRuleSetting.compareName(
+									NAME_KEY_OBJECT_FIELD_ID) &&
+							  objectValidationRuleSetting.compareName(
+								  ObjectValidationRuleSettingConstants.
+									  NAME_OUTPUT_OBJECT_FIELD_ID))) {
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/converter/ObjectValidationRuleDTOConverter.java:139]: Do we need to also check NAME_OUTPUT_OBJECT_FIELD_ID here? It seems that it's already covered in the if condition below. We should keep NAME_OUTPUT_OBJECT_FIELD_ID working as before, even without this feature flag.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java [hunk 3]
```diff
@@ -538,12 +538,25 @@ else if (Objects.equals(
 		for (ObjectValidationRuleSetting objectValidationRuleSetting :
 				objectValidationRuleSettings) {
 
-			if (!(objectValidationRuleSetting.compareName(
+			if (FeatureFlagManagerUtil.isEnabled("LPS-187854") &&
+				!(objectValidationRuleSetting.compareName(
 					ObjectValidationRuleSettingConstants.
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:543]: Do we need to also check NAME_KEY_OBJECT_FIELD_ID here? It seems that it's already covered in the if condition below. We should keep NAME_KEY_OBJECT_FIELD_ID working as before, even without this feature flag.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java [hunk 4]
```diff
@@ -529,19 +531,32 @@ else if (Objects.equals(
...
 		for (ObjectValidationRuleSetting objectValidationRuleSetting :
 				objectValidationRuleSettings) {
 
+			if (FeatureFlagManagerUtil.isEnabled("LPS-187854") &&
+				!(objectValidationRuleSetting.compareName(
+					ObjectValidationRuleSettingConstants.
+						NAME_KEY_OBJECT_FIELD_ID) ||
+				  objectValidationRuleSetting.compareName(
+					  ObjectValidationRuleSettingConstants.
+						  NAME_OUTPUT_OBJECT_FIELD_ID))) {
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:547]: Do we need to also check NAME_OUTPUT_OBJECT_FIELD_ID here? It seems that it's already covered in the if condition below. We should keep NAME_OUTPUT_OBJECT_FIELD_ID working as before, even without this feature flag.

**Comment by guilhermedcamacho** [comment]
ci:stop

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/converter/ObjectValidationRuleDTOConverter.java (hunk 2 above)
Comment (mateussgn) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/converter/ObjectValidationRuleDTOConverter.java:139]: Yes, we need to keep the antique behavior.
After the Feature Flag removal we will have the follow snippet:
```java
if (!(objectValidationRuleSetting.compareName(
	        ObjectValidationRuleSettingConstants.
		        NAME_KEY_OBJECT_FIELD_ID) ||
	objectValidationRuleSetting.compareName(
		ObjectValidationRuleSettingConstants.
			NAME_OUTPUT_OBJECT_FIELD_ID)))
``` 
Indicating that the setValue only accepts those two types of  `objectValidationRuleSetting` names

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:sf

**Review by guilhermedcamacho — APPROVED** [review]

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectValidationRuleResourceImpl.java [hunk 5]
```diff
@@ -261,6 +262,28 @@ protected void preparePatch(
 					getObjectValidationRuleSettings()));
 	}
 
+	private com.liferay.object.model.ObjectValidationRuleSetting
+			_setObjectValidationRuleSettingProperties(
+				com.liferay.object.model.ObjectValidationRuleSetting
+					serviceBuilderObjectValidationRuleSetting,
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectValidationRuleResourceImpl.java:268]: Hey @mateussgn , I'm just wondering if we should order this variable here, maybe as it starts with s, it should come after objectDefinitionId

Comment (mateussgn) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectValidationRuleResourceImpl.java]: Sure, you are right!

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:forward:force

**Comment by guilhermedcamacho** [comment]
ci:forward:force

**Comment by mateussgn** [comment]
ci:forward:force

**Pull request closed without merging**